create_images.sh

```diff
@@ -13,19 +13,21 @@ _os_version=10
 
 echo "==> Generating images for patch level $_os_patch_level"
 
+# True by default, analogous to BOARD_USES_RECOVERY_AS_BOOT:
+[ "$_recovery_ramdisk" = "false" ] && _ramdisk=ramdisk.img || _ramdisk=ramdisk-recovery.img
+_ramdisk=$ANDROID_ROOT/out/target/product/$_device/$_ramdisk
+
+[ ! -f "$_ramdisk" ] && echo "WARNING: $_ramdisk does not exist!"
+
 # mkdir -p $(dirname $_boot_out)
 
 if [[ "$_has_dtbo" == "true" ]]; then
     _dts_folder="$_out/arch/arm64/boot/dts"
     _files=$(find "$_dts_folder" -iname "*.dtbo")
     echo "==> Creating dtboimg from $_files"
     _mkdtboimg="$_kernel_path/scripts/mkdtboimg.py"
-    if [[ ! -f "$_mkdtboimg" ]]; then
-        _mkdtboimg="$ANDROID_ROOT/prebuilts/misc/linux-x86/libufdt/mkdtimg"
-    fi
-    if [[ ! -f "$_mkdtboimg" ]]; then
-        _mkdtboimg="$ANDROID_ROOT/system/libufdt/utils/src/mkdtboimg.py"
-    fi
+    [[ ! -f "$_mkdtboimg" ]] && _mkdtboimg="$ANDROID_ROOT/prebuilts/misc/linux-x86/libufdt/mkdtimg"
```

Comment:
Wouldn't it be more clear to add another `if` condition if `$ANDROID_ROOT/prebuilts/misc/linux-x86/libufdt/mkdtimg` doesn't exist?

Seeing that we have 3 possiblities, in my opinion an `if .. elif .. elif` dance could be even more clear.

Reply:
That might be more clear but requires the path to be repeated twice: once in `[ ! -f "the/path" ]`, another time in `_mkdtboimg="the/path"` assignment. Added an extra condition if the final assignment doesn't yield a usable script.